temporal-sdk/src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -83,11 +86,19 @@
     this.interceptors = Objects.requireNonNull(interceptors);
   }
 
-  private void addActivityImplementation(
+  private void registerActivityImplementation(
       Object activity, BiFunction<Method, Object, ActivityTaskExecutor> newTaskExecutor) {
     if (activity instanceof Class) {
       throw new IllegalArgumentException("Activity object instance expected, not the class");
     }
+    if (activity instanceof DynamicActivity) {
+      if (dynamicActivity != null) {
+        throw new IllegalStateException(
+            "An implementation of DynamicActivity is already registered with the worker");
+      }
```

Comment:
Would be good to indicate workers can only have one `DynamicActivity` implementation in a docstring somewhere